cql3/statements/describe_statement.hh

```diff
@@ -36,20 +36,37 @@ namespace cql3 {
 
 class query_processor;
 
-// Represents description of keyspace_element
+/// Type representing an entity that can be restored by performing
+/// a SINGLE CQL query. It can correspond to a tangible object such as
+/// a keyspace, a table, or a role, as well as to a more abstract concept
+/// like a role grant.
+///
+/// Instances of this type correspond to the output of `DESCRIBE` statements.
+///
+/// ! Important note: !
+/// -------------------
+/// `description` does NOT perform any additional formatting. As a result,
+/// code creating instances of it IS REQUIRED to make sure all of the fields
+/// represent valid and correct identifiers/CQL statement. That, for example,
+/// encompasses wrapping an identifier with quotation marks if it contains
+/// a whitespace character, an uppercase letter, or a quotation mark. Other edge
+/// cases may also be possible. The user of this interface is responsible for
+/// making sure they take into considerations all relevant circumstances and test
+/// their implementation.
+/// -------------------
+///
+/// See scylladb/scylladb#11106 and scylladb/scylladb#18750 for more context.
 struct description {
-    sstring _keyspace;
-    sstring _type;
-    sstring _name;
-    std::optional<sstring> _create_statement;
-
-    // Description without create_statement
-    description(replica::database& db, const data_dictionary::keyspace_element& element);
-
-    // Description with create_statement
-    description(replica::database& db, const data_dictionary::keyspace_element& element, bool with_internals);
-
-    description(replica::database& db, const data_dictionary::keyspace_element& element, sstring create_statement);
+    /// The name of the keyspace the entity belongs to.
+    /// Empty optional if and only if the entity does not belong to any keyspace.
+    std::optional<sstring> keyspace;
+    /// The name of the type of an entity, e.g. a role may be of type: role
+    sstring type;
+    /// The name of the entity itself, e.g. a keyspace of name `ks` will be of name: ks
+    sstring name;
+    /// CQL statement that can be used to restore the entity.
+    /// Empty in special cases.
+    std::optional<sstring> create_statement;
 
     std::vector<bytes_opt> serialize() const;
```

Comment:
How does serialize() work if it's generic?

Reply:
Could you elaborate? Do you mean I should describe the semantics of `serialize()` here?